src/hotspot/share/classfile/classLoaderData.cpp

```diff
@@ -949,8 +949,8 @@ void ClassLoaderData::print_on(outputStream* out) const {
     _holder.print_on(out);
     out->print_cr("");
   }
-  out->print_cr(" - class loader        %p", _class_loader.ptr_raw());
-  out->print_cr(" - metaspace           %p", _metaspace);
+  out->print_cr(" - class loader        " PTR_FORMAT, p2i(_class_loader.ptr_raw()));
```

Comment:
I'm surprised the p2i's were added. If the values are pointers then p2i should not be needed. If using p2i then INTPTR_FORMAT is the correct format specifier to use.

Reply:
I'm not sure the original intentions of these two macros. It looks like the definitions of PRT_FORMAT and INTPTR_FORMAT are identical:

https://github.com/openjdk/jdk/blob/c3abdc9aadc734053dbcc43d5294d5f16a0b0ce3/src/hotspot/share/utilities/globalDefinitions.hpp#L129-L132

Also I find many occurrences that using p2i while format specifier is PRT_FORMAT. If this is indeed wrong, I may fix them later.

Reply:
Even if a pointer is given, p2i is necessary for PRT_FORMAT and INTPTR_FORMAT specifier,  the compiler will throw errors without p2i.